lib/common/memory/src/mmap_type.rs

```diff
@@ -432,26 +434,36 @@ pub enum Error {
 /// # Panics
 ///
 /// - panics when the mmap data is not correctly aligned for type `T`
-unsafe fn mmap_to_type_unbounded<'unbnd, T>(mmap: &mut MmapMut) -> Result<&'unbnd mut T>
+unsafe fn mmap_prefix_to_type_unbounded<'unbnd, T>(mmap: &mut MmapMut) -> Result<&'unbnd mut T>
 where
     T: Sized,
 {
     let size_t = mem::size_of::<T>();
 
     // Assert size
-    if mmap.len() != size_t {
-        return Err(Error::SizeExact(size_t, mmap.len()));
+    if mmap.len() < size_t {
+        return Err(Error::SizeLess(size_t, mmap.len()));
     }
 
     // Obtain unbounded bytes slice into mmap
     let bytes: &'unbnd mut [u8] = {
         let slice = mmap.deref_mut();
-        slice::from_raw_parts_mut(slice.as_mut_ptr(), slice.len())
+        slice::from_raw_parts_mut(slice.as_mut_ptr(), size_t)
     };
 
     // Assert alignment and size
     assert_alignment::<_, T>(bytes);
-    debug_assert_eq!(mmap.len(), bytes.len());
+
+    #[cfg(debug_assertions)]
+    if mmap.len() != size_t {
+        log::warn!(
+            "Mmap length {} is not equal to size of type {}",
+            mmap.len(),
+            size_t,
+        );
+    }
+
+    #[cfg(debug_assertions)]
     if bytes.len() != mem::size_of::<T>() {
```

Comment:
This check never fails. If we want to return `Error::SizeExact` on debug builds, we need to adjust the check:
```suggestion
    if mmap.len() != mem::size_of::<T>() {
```

Reply:
Then it would be the same as a previous check. 
It is kinda redundant, but may serve explanation functionality

Reply:
After discussion, decided to keep it as is.